lib/grape/validations/validators/coerce.rb

```diff
@@ -68,9 +68,10 @@ def coerce_value(val)
         # define default values for structures, the dry-types lib which is used
         # for coercion doesn't accept nil as a value, so it would fail
         if val.nil?
-          return []      if type == Array || type.is_a?(Array)
+          return []      if type == Array
           return Set.new if type == Set
           return {}      if type == Hash
+          return val     if type.is_a?(Array)
```

Comment:
I'm not sure about this `type.is_a?(Array)` check. I assume that passing a `nil` value to a parameter with multiple types might lead to ambiguity?

Reply:
Unfortunately, it was necessary
```ruby
    it 'handles default value to array' do
      subject.params do
        optional :a, type: Array[Integer], default: []
      end
      subject.put('/test') { declared(params).to_json }
      put '/test', { a: nil }
      expect(last_response.status).to eq(200)
      expect(last_response.body).to eq('{"a":[]}')
    end
```
That test fails because `type != array` but it's `type.is_a?(Array)
I'll make a PR to fix it